### PR TITLE
Include more documentations

### PR DESCRIPTION
**Pull request by TongkunZhang**
- [x] Use docsify for documentation
- [x] Clean up outdated documentations
  - [x] Installation
  - [x] Prerequisties
  - [x] getting-started/sample-build
  - [x] convert-source-code
  - [x] Fix old hyperlinks
   - [x] README.md
- [x] Add more documentations
  - [x] CLI References
    - [x] makei
    - [x] crtfrmstmf
  - [ ] Integrations
  - [x] Build RPM
  - [ ] Recipes
  - [ ] Changelogs
    - [ ] v2 changes
  - [ ] Source Structure
- [ ] Use github pages for documentation hosting
- [ ] Deprecated Wiki

### CODE REVIEW

File: docs/prepare-the-project/rules.mk.md [hunk 1]
```diff
@@ -40,7 +44,7 @@ This line at the project root directory will tell Bob to build all the subdirect
 
 The object list section is where you list out each object Bob should build, grouped by object type and separated by spaces.  Object names should be specified as they appear in the Integrated File System (IFS), so program names end in `.PGM`, files end in `.FILE`, etc.  All object names should be in upper case.  For ease of maintenance, it is recommended that the objects are listed alphabetically.  Here's an example of part of an object list section:
```
Comment (edmundreinhardt) [inline docs/prepare-the-project/rules.mk.md:45]: Object names should be specified as the unqualified IFS file name.  For example program names would be `MYPGM.PGM` and files  would be 'MYFILE.FILE', etc.

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]
Do we need this empty file or should it be .gitignore d?
docs/.nojekyll

**Comment by TongkunZhang** [comment]
> Do we need this empty file or should it be .gitignore d? docs/.nojekyll

It is needed for [Bypassing Jekyll on GitHub Pages](https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/)

File: docs/getting-started/sample-build.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+   Build Completed!
+  ```
+
+  * all of the output is stored in `.logs/output.log`
+  * all of the job logs are gathered in '.logs/joblog.json` and can be viewed with any JSON viewer
+  * the event files for all compiles are gathered under the `.evfevent` directory
+
+## The sample build process in action (quadruple speed)
+
+![Sample Build Demo](sample-build.assets/sample-build.gif)
```
[the remark this answers is not in the document]
Comment (TongkunZhang) [reply docs/getting-started/sample-build.md]: going to rerecord this after the example is clean

**Review by edmundreinhardt — APPROVED** [review]
Fantastic work!!!

**Pull request merged**